### PR TITLE
325 refactor etlapi for use in jobtracker v2

### PR DESCRIPTION
**Pull request by bossie**
Implements https://github.com/Open-EO/openeo-geopyspark-driver/issues/325.

### CODE REVIEW

**Comment by bossie** [comment]
Points 1-3 of https://github.com/Open-EO/openeo-geopyspark-driver/issues/325 are implemented by encapsulating the job costs calculation (in this case, the ETL API).

Do you think this might work @soxofaan ?

File: openeogeotrellis/job_tracker_v2.py [hunk 1]
```diff
@@ -68,6 +80,72 @@ def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetada
         raise NotImplementedError
 
 
+class JobCostsCalculator(metaclass=abc.ABCMeta):
+    @abc.abstractmethod
+    def calculate_costs(self, job_info: dict, job_metadata: _JobMetadata, result_metadata: dict) -> float:
+        raise NotImplementedError
+
+
+class EtlApiJobCostsCalculator(JobCostsCalculator):
```
Comment (soxofaan) [inline openeogeotrellis/job_tracker_v2.py:89]: If I understand correctly this in not only about calculating the cost, but also reporting/logging, right?

File: openeogeotrellis/job_tracker_v2.py [hunk 2]
```diff
@@ -68,6 +80,72 @@ def get_job_metadata(self, job_id: str, user_id: str, app_id: str) -> _JobMetada
...
+    def __init__(self, etl_api: EtlApi, etl_api_access_token: str):
+        self._etl_api = etl_api
+        self._etl_api_access_token = etl_api_access_token
+
+    def calculate_costs(self, job_info: dict, job_metadata: _JobMetadata, result_metadata: dict) -> float:
```
Comment (soxofaan) [inline openeogeotrellis/job_tracker_v2.py:94]: It feels there is some conceptual redundancy between `job_info` and `job_metadata`. 
Is is easily possible to eliminate one in favor of the other?

File: openeogeotrellis/job_tracker_v2.py (hunk 1 above)
Comment (soxofaan) [reply openeogeotrellis/job_tracker_v2.py:89]: Another thing: as far as I understand, the job cost calculation is quite separate from the job tracker itself. Could it make sense to move it  to its own module to have clearer separation of concerns?

**Review by soxofaan — COMMENTED** [review]
overall this looks good I think

File: openeogeotrellis/job_tracker_v2.py (hunk 1 above)
Comment (bossie) [reply openeogeotrellis/job_tracker_v2.py:89]: > If I understand correctly this in not only about calculating the cost, but also reporting/logging, right?

Yes: you give the ETL API some resource usage and it returns the costs in credits; then you patch those costs in the batch job entity.

> Another thing: as far as I understand, the job cost calculation is quite separate from the job tracker itself. Could it make sense to move it to its own module to have clearer separation of concerns?

Possibly. There's also the fact that a custom `JobCostsCalculator` implementation relies on the private `_JobMetadata` class so there's definitely room for improvement.

File: openeogeotrellis/job_tracker_v2.py (hunk 2 above)
Comment (bossie) [reply openeogeotrellis/job_tracker_v2.py:94]: Also possibly. This was very ad-hoc to get things started.

File: openeogeotrellis/job_tracker_v2.py (hunk 1 above)
Comment (bossie) [reply openeogeotrellis/job_tracker_v2.py]: I moved it to the `job_costs_calculator` module.

**Comment by bossie** [comment]
Could take a quick look at this again @soxofaan ?

**Pull request merged**